rootfs/api/models/build.py

```diff
@@ -52,8 +52,8 @@ def version(self):
         return 'git-{}'.format(self.sha) if self.source_based else 'latest'
 
     def create(self, user, *args, **kwargs):
-        latest_release = self.app.release_set.latest()
-
+        latest_release = self.app.release_set.filter(failed=False).latest()
```

Comment:
I still feel like we'd want to create a new release using the latest release's config and version, regardless if it failed to deploy or not. I feel like this is an unintended breaking change for deploying apps.

e.g. given this situation:

 - I deploy v2 successfully
 - I deploy v3 successfully with DATABASE_URL
 - I deploy v4 with DATADOG_API_KEY, but the deploy fails due to flaky networking between datadog and the app
 - I deploy v5 successfully

In this situation given the following code, we'd lose DATADOG_API_KEY in v5. I don't feel that's the intention here.